create-fields fails when trying to create a module

When I try to run the ```make create-fields MODULE=curator FILESET=log``` command I get this error:

```
panic: interface conversion: interface {} is []interface {}, not string

goroutine 1 [running]:
main.accumulateRemoveFields(0x1109640, 0xc4200867e0, 0x0, 0x0, 0x0, 0x1, 0x1, 0x1)
	/Users/dante/go/src/github.com/elastic/beats/filebeat/scripts/generator/fields/main.go:158 +0x205
main.getProcessors(0xc42000a1e0, 0x4, 0x4, 0x7ffeefbffade, 0x35, 0x10ffea0)
	/Users/dante/go/src/github.com/elastic/beats/filebeat/scripts/generator/fields/main.go:220 +0x24e
main.(*pipeline).toFieldsYml(0xc42000e380, 0x0, 0x7ffeefbffb1d, 0x7, 0x7ffeefbffb2f, 0x3, 0xc42000e380)
	/Users/dante/go/src/github.com/elastic/beats/filebeat/scripts/generator/fields/main.go:291 +0x45
main.main()
	/Users/dante/go/src/github.com/elastic/beats/filebeat/scripts/generator/fields/main.go:342 +0x246
exit status 2
make: *** [create-fields] Error 1



I think it has something to do with the pipeline, but it appears correct:

{
  "description": "Pipeline for parsing curator log logs",
  "processors": [
    {
      "json": {
        "field": "message"
      }
    },
    {
      "grok": {
        "field": "message",
        "patterns": [
            "Action ID: %{INT:action_id}"
        ],
        "ignore_missing": true
      }
    },
    {
      "rename": {
        "field": "loglevel",
        "target_field": "level"
      }
    },
    {
      "remove": {
        "field": [
            "prospector.type"
        ]
      }
    }
    ],
  "on_failure" : [{
    "set" : {
      "field" : "error.message",
      "value" : "{{ _ingest.on_failure_message }}"
    }
  }]
}

--- a/filebeat/scripts/generator/fields/main.go
+++ b/filebeat/scripts/generator/fields/main.go
@@ -155,8 +155,22 @@ func accumulatePatterns(grok interface{}) ([]string, error) {
 func accumulateRemoveFields(remove interface{}, out []string) []string {
 	for k, v := range remove.(map[string]interface{}) {
 		if k == "field" {
-			vs := v.(string)
-			return append(out, vs)
+			switch vs := v.(type) {
+			case string:
+				return append(out, vs)
+			case []string:
+				for _, vv := range vs {
+					out = append(out, vv)
+				}
+			case []interface{}:
+				for _, vv := range vs {
+					vvs := vv.(string)
+					out = append(out, vvs)
+				}
+			default:
+				return out
+
+			}
 		}
 	}
 	return out

--- a/filebeat/scripts/generator/fields/main_test.go
+++ b/filebeat/scripts/generator/fields/main_test.go
@@ -12,6 +12,11 @@ type FieldsGeneratorTestCase struct {
 	fields   []*fieldYml
 }
 
+type RemoveProcessorTestCase struct {
+	processor map[string]interface{}
+	fields    []string
+}
+
 func TestFieldsGenerator(t *testing.T) {
 	tests := []FieldsGeneratorTestCase{
 		FieldsGeneratorTestCase{
@@ -157,5 +162,40 @@ func TestFieldsGeneratorKnownLimitations(t *testing.T) {
 
 		f := generateFields(fs, false)
 		assert.False(t, reflect.DeepEqual(f, tc.fields))
+	}
+}
+
+func TestRemoveProcessor(t *testing.T) {
+	tests := []RemoveProcessorTestCase{
+		RemoveProcessorTestCase{
+			processor: map[string]interface{}{
+				"field": []string{},
+			},
+			fields: []string{},
+		},
+		RemoveProcessorTestCase{
+			processor: map[string]interface{}{
+				"field": []interface{}{},
+			},
+			fields: []string{},
+		},
+		RemoveProcessorTestCase{
+			processor: map[string]interface{}{
+				"field": "prospector.type",
+			},
+			fields: []string{"prospector.type"},
+		},
+		RemoveProcessorTestCase{
+			processor: map[string]interface{}{
+				"field": []string{"prospector.type", "input.type"},
+			},
+			fields: []string{"prospector.type", "input.type"},
+		},
+	}
+
+	for _, tc := range tests {
+		out := []string{}
+		res := accumulateRemoveFields(tc.processor, out)
+		assert.True(t, reflect.DeepEqual(res, tc.fields))
 	}
 }

